# HW5 

## Q2

Datasets are often very large. To get to know them, we seldom read it compeletely, instead, we get samples from the dataset and use samples to estimate the perform of the dataset.So when testing the outcome, we use the population parameter that is the "correct answer" to ensure the outcome fits our dataset.


$ x_i $: Individual data points, like a person’s height or income.

$  \bar{x} $: The average of these data points in the sample.

$  \mu $: The true average of the entire population, which we don’t usually know.

$  \mu_0 $: A specific value we’re testing, like “is the population average equal to 50?”


So, the test results help us draw conclusions about $ \mu $ (the population’s true average), not just the average $\bar{x} $ from our small sample.

## Q3

We want to show that the null hypothese is False, while its hard to deny directly. We first assume it is true and calculate the p-value then we see the possibility of null hypothese to be true.

## Q4

When we calculate a p-value, we’re comparing the observed test statistic to the sampling distribution of the test statistic under the assumption that the null hypothesis is true. A small p-value means our observed test statistic is far from the typical values we’d expect if the null hypothesis were true, suggesting that the observed result is unusual and doesn't fit well with the null hypothesis. 

## Q5

In [1]:
import numpy as np

# Given data
n_couples = 124
observed_right_tilts = 80
null_probability = 0.5
n_simulations = 10000

# Simulate coin flips under the null hypothesis
simulated_right_tilts = np.random.binomial(n=n_couples, p=null_probability, size=n_simulations)

# Calculate p-value (proportion of simulations where the number of "right-tilts" >= observed 80)
p_value = np.mean(simulated_right_tilts >= observed_right_tilts)

# Print the p-value
print(f"P-value: {p_value}")


P-value: 0.0008


## Q6

a smaller p-value makes the null hypothesis less plausible, but it never definitively proves it false. The decision to reject the null depends on context, the p-value threshold used, and the overall design of the study.

## Q8

Null Hypothesis $(H_0)$: The probability of correctly state which was poured first is 50/50, or p=0.5. So, any deviation from 50% is due to random chance.

Alternating hypothesis$(H_A)$: This tests whether the probability is different from 50%, in either direction (greater or less than 50%), or p≠0.5

Observed Proportion: In the real data, 49 out of a sample of 80 students are able to correctly state which was poured first, or 61.25% able to correctly state which was poured first.

In [3]:
import numpy as np
from scipy.stats import norm

# Given values
p_null = 0.5  # Null hypothesis proportion (50%)
correct_answers = 49  # Number of correct answers
n = 80  # Total number of students

# Calculate the observed proportion
observed_p = correct_answers / n  # Proportion of correct answers

# Calculate standard error
standard_error = np.sqrt((p_null * (1 - p_null)) / n)

# Calculate z-score
z_score = (observed_p - p_null) / standard_error

# Calculate p-value (two-tailed test)
p_value = 2 * (1 - norm.cdf(abs(z_score)))

# Print results
print(f"Observed Proportion: {observed_p:.4f}")
print(f"Z-score: {z_score:.4f}")
print(f"P-value: {p_value:.4f}")


Observed Proportion: 0.6125
Z-score: 2.0125
P-value: 0.0442


|p-value|Evidence|
|-|-|
|$$p > 0.1$$|No evidence against the null hypothesis|
|$$0.1 \ge p > 0.05$$|Weak evidence against the null hypothesis|
|$$0.05 \ge p > 0.01$$|Moderate evidence against the null hypothesis|
|$$0.01 \ge p > 0.001$$|Strong evidence against the null hypothesis|
|$$0.001 \ge p$$|Very strong evidence against the null hypothesis|

According to the table, there is moderate evidence against the null hypothesis, suggesting a real difference in the ability to identify which was poured first. 

## Chatbot History



1. **Hypothesis Testing Basics**: We discussed the distinctions between the null hypothesis (\(H_0\)) and alternative hypothesis (\(H_A\)), explaining their definitions and roles in hypothesis testing.

2. **Specific Example**:
   - You provided a specific scenario where the null hypothesis is that the probability of correctly stating which was poured first (tea or milk) is 50% (\(p = 0.5\)).
   - The alternative hypothesis was defined as \(H_A: p \neq 0.5\).

3. **Observed Proportion**: You mentioned that in your sample of 80 students, 49 were able to correctly state which was poured first, leading to an observed proportion of 61.25%.

4. **Hypothesis Test Setup**:
   - We outlined the steps for conducting a hypothesis test for this scenario, including calculating the standard error, z-score, and p-value.

5. **Calculations**:
   - I performed the calculations for the z-score and p-value based on the provided data, resulting in a z-score of approximately 2.01 and a p-value of around 0.044.

6. **Interpretation of Results**:
   - We interpreted the results, concluding that since the p-value was less than the common significance level of 0.05, we could reject the null hypothesis, indicating a significant difference in the ability of students to identify which was poured first.

7. **Code for Hypothesis Testing**:
   - You requested the Python code to perform the calculations, which I provided along with a detailed explanation of each step in the code.



https://chatgpt.com/c/6704411f-a1e4-8004-90c0-e41f88e72c01